### DISCUSSION TITLE
Keycloak as Service Provider in SAML Federations (with single ACS endpoint)

### DISCUSSION THREAD

**Discussion by maeni70** [keycloak/keycloak#20898] [Ideas]
Hello

I am evaluating Keycloak right now as an identity provider for an application using OIDC. And we would like register Keycloak as a service provider in eduGAIN (academic SAML federation).

There is just one catch. The ACS endpoints are specific to each registered identity provider (by the "alias"):
https://keycloak.server/auth/realms/realmname/broker/alias/endpoint

Our case is special that we would like to have features like automatic redirection to IdPs (skipping login form), which works in Keycloak, but probably not so easy to implement when using an SP Proxy approach as a workaround (e.g. SATOSA with discovery services).

IMO, there would be many use cases which would benefit from single ACS endpoint in realm.

Cheers,
Roger

**Comment by anussba** [comment]
Hi
We also would be very interested in this feature, it is a pain that keycloak is not able to provide a single ACS endpoint (for many IDPs) per realm. (IDP Discovery)
So when you want to join large SAML Federations like eduGAIN you have to set up
a saml proxy like satosa in front of keycloak in order to do so.
Specially in the education/universties it would be a gamechanger if we could do without a proxy solution.

Thanks
br
Andreas

**Comment by CGMBenjamin** [comment]
Me too!

I'm try to explain my scenario:

Target: Get a Token without any login screen on my Springboot develops via API passing username and password in a http request to a Client on Keycloak (used as Service Provider) and configured an IdP conection to the final users repository using SAML2 protocol.

Does anybody any idea to achieve it?

Thanks in advanced.

**Comment by samuelpetrosyan** [comment]
Any updates regarding this?
